packages/flutter/lib/src/material/navigation_drawer.dart

```diff
@@ -328,7 +328,7 @@ class _NavigationDestinationBuilder extends StatelessWidget {
           child: InkWell(
             highlightColor: Colors.transparent,
             onTap: info.onTap,
-            borderRadius: const BorderRadius.all(Radius.circular(28.0)),
+            customBorder: info.indicatorShape ?? navigationDrawerTheme.indicatorShape ?? defaults.indicatorShape!,
```

Comment:
Wow, we really appear to have dropped the ball here. Nice fix.

Reply:
Thanks @HansMuller, and naah probably on purpose, just to see if I'm awake enough to notice, and give me an easy fix 😃